indexer/src/main.rs

```diff
@@ -37,30 +41,39 @@ fn run(home_dir: std::path::PathBuf, config: RunConfigArgs) -> Result<()> {
     } else {
         None
     };
-
+    // firat: added some logic to handle the case when fastnear is enabled. Needs tests and maybe refactoring for base case
     // TODO: refactor
     let block_res = system.block_on(async move {
-        let mut indexer = IndexerWrapper::new(indexer_config, addresses_to_rollup_ids);
-        if let Some(_) = config.metrics_ip_port_address {
-            indexer.enable_metrics(registry.clone())?;
-        }
-
-        let (view_client, _) = indexer.client_actors();
-        let (block_handle, candidates_stream) = indexer.run();
-        let mut candidates_validator = CandidatesValidator::new(view_client);
-        if let Some(_) = config.metrics_ip_port_address {
-            candidates_validator.enable_metrics(registry.clone())?;
+        let validated_stream: Receiver<PublishData>;
+        if cfg!(feature = "use_fastnear") {
+            let fastnear_indexer = FastNearIndexer::new(addresses_to_rollup_ids);
+            validated_stream = fastnear_indexer.run();
+        } else {
```

Comment:
I wouldnt expect this to compile, but if not,

```suggestion
        cfg!(feature = "use_fastnear") 
        {
            let fastnear_indexer = FastNearIndexer::new(addresses_to_rollup_ids);
            validated_stream = fastnear_indexer.run();
        }
```

Or there's also `cfg_if` crate